pilot/pkg/config/memory/controller.go

```diff
@@ -103,7 +103,7 @@ func (c *controller) UpdateStatus(config config.Config) (newRevision string, err
 }
 
 func (c *controller) Patch(orig config.Config, patchFn config.PatchFunc) (newRevision string, err error) {
-	cfg := patchFn(orig.DeepCopy())
+	cfg, _ := patchFn(orig.DeepCopy())
```

Comment:
do we not need the patch type here? I would think it is still relevant (and needed for testing) and we should be able to perform all the different types locally (using jsonpath library)?

Reply:
For now I am just changing this to err for a patch type we don't support. We will have to expose some of the generated code from crdclient to convert config.Config into a JSON marshallable/patchable struct.